3.9/graphs-pregel-algorithms.md

```diff
@@ -0,0 +1,298 @@
+iteration by using a HyperLogLog counter with 64 buckets. This should work well
+on large graphs and on smaller ones as well. The memory requirements should be
+**O(n * d)** where *n* is the number of vertices and *d* the diameter of your
+graph. Each vertex will store a counter for each iteration of the algorithm.
+
+The algorithm can be used like this:
+
+```js
+const pregel = require("@arangodb/pregel");
+const handle = pregel.start("effectivecloseness", "yourgraph", {resultField: "closeness"});
+```
+
+#### LineRank
+
+Another common measure is the [betweenness* centrality](https://en.wikipedia.org/wiki/Betweenness_centrality){:target="_blank"}:
+It measures the number of times a vertex is part of shortest paths between any
+pairs of vertices. For a vertex *v* betweenness is defined as:
+
+![Vertex Betweenness Formula](images/betweenness.png)
+
+Where the &sigma; represents the number of shortest paths between *x* and *y*,
+and &sigma;(v) represents the number of paths also passing through a vertex *v*.
+By intuition a vertex with higher betweenness centrality will have more
+information passing through it.
+
+**LineRank** approximates the random walk betweenness of every vertex in a
+graph. This is the probability that someone starting on an arbitrary vertex,
+will visit this node when he randomly chooses edges to visit.
+
+The algorithm essentially builds a line graph out of your graph
+(switches the vertices and edges), and then computes a score similar to PageRank.
+This can be considered a scalable equivalent to vertex betweenness, which can
+be executed distributedly in ArangoDB. The algorithm is from the paper
+*Centralities in Large Networks: Algorithms and Observations (U Kang et.al. 2011)*.
+
+```js
+const pregel = require("@arangodb/pregel");
+const handle = pregel.start("linerank", "yourgraph", {resultField: "linerank"});
+```
+
+### Community Detection
+
+Graphs based on real world networks often have a community structure.
+This means it is possible to find groups of vertices such that each vertex
+group is internally more densely connected than outside the group.
+This has many applications when you want to analyze your networks, for example
+Social networks include community groups (the origin of the term, in fact)
+based on common location, interests, occupation, etc.
+
+#### Label Propagation
+
+*Label Propagation* can be used to implement community detection on large
+graphs. The algorithm assigns a community, more precisely, a Community ID 
+(a natural number), to every vertex in the graph. 
+The idea is that each vertex should be in the community that most of
+its neighbors are in. 
+
+At first, the algorithm assigns unique initial Community IDs to the vertices. 
+There is no guarantee that a vertex obtains the same initial 
+ID in two different runs of the algorithm, even if the graph does not change
+(although, it may often happen). Moreover, there is no guarantee on a particular
```

Comment:
```suggestion
ID in two different runs of the algorithm, even if the graph does not change 
(because the sharding may change). Moreover, there is no guarantee on a particular
```